src/AnchoredOverlay/AnchoredOverlay.tsx

```diff
@@ -0,0 +1,134 @@
+import React, {useCallback, useEffect, useMemo, useRef, useState} from 'react'
+import Overlay from '../Overlay'
+import Button, {ButtonProps} from '../Button'
+import {useFocusTrap} from '../hooks/useFocusTrap'
+import {useFocusZone} from '../hooks/useFocusZone'
+import {useAnchoredPosition, useRenderForcingRef} from '../hooks'
+import {uniqueId} from '../utils/uniqueId'
+
+export interface AnchoredOverlayProps {
+  /**
+   * A custom function component used to render the anchor element.
+   * Will receive the selected text as `children` prop when an item is activated.
+   * Uses a `DropdownButton` by default.
+   */
+  renderAnchor?: <T extends React.HTMLAttributes<HTMLElement>>(props: T) => JSX.Element
+
+  /**
+   * Determines whether the overlay portion of the component should be shown or not
+   */
+  open: boolean
+
+  /**
+   * A callback which is called whenever the overlay is currently closed and an "open gesture" is detected.
+   */
+  onOpen?: (gesture: 'anchor-click' | 'anchor-key-press') => unknown
+
+  /**
+   * A callback which is called whenever the overlay is currently open and a "close gesture" is detected.
+   */
+  onClose?: (gesture: 'click-outside' | 'escape') => unknown
+}
+
+/**
+ * An `AnchoredOverlay` provides an anchor (button by default) that will open a floating overlay.
+ * The overlay can be opened and navigated using keyboard or mouse.
+ */
+export const AnchoredOverlay: React.FC<AnchoredOverlayProps> = ({
+  renderAnchor = <T extends ButtonProps>(props: T) => <Button {...props} />,
```

Comment:
Is this a useful default? Won't the consumer need to override this default in almost every case?

Reply:
Yeah, that's fair.  The other components (DropdownMenu, ActionMenu) can specify sane defaults and it can be required here.